apps/system/js/haiku_emoji_panel.js

```diff
@@ -55,18 +67,27 @@
     },
 
     emojiClickHandler: function(e) {
-      var target = e.target;
-      var receiver = window.app.pairNumber;
-      var messageBody = target && target.dataset.icon;
-      var audio = document.querySelector('#sms_tone');
-      audio.play();
+      //if it has been >= 2 seconds since last msg, continue sending msg
+      if (handleNextSMS) {
+        var target = e.target;
+        var receiver = window.app.pairNumber;
+        var messageBody = target && target.dataset.icon;
+        var audio = document.querySelector('#sms_tone');
+        audio.play();
+        handleNextSMS = false;
+        trueAfterDelay(2000);        
 
       if (!messageBody) {
         return;
       }
 
       console.log('Haiku:sending emoji ', messageBody);
       mozMessage.send(receiver, messageBody);
+      }
+      //if it's < 2 seconds, disable hover/pointer (for animation) and do not send msg
+      document.getElementById('emoji_left').style.pointerEvents = 'none';
+      document.getElementById('emoji_middle').style.pointerEvents = 'none';
+      document.getElementById('emoji_right').style.pointerEvents = 'none';
```

Comment:
A css class e.g.  .avoid-clicks {
  pointer-events: none;
} can be added here. We can try adding it to parent section with id emoji-icons